### PR TITLE
Remove strtok(3)

### PR DESCRIPTION
**Pull request by nabijaczleweli**
### Motivation and Context
I have many words to spill about strtok(3), especially when used in libraries in multi-threaded contexts, unfortunately they've been replaced by an import path.

### Description
It's mostly what it says on the can. See individual commit messages for where it's more.

I also normalised, where applicable, to `for (s = strtok_r(str); s != NULL, s = strtok_r(NULL))`. Initialising `tmp` to NULL will, if the string is also NULL (e.g. due to a failed strdup(), which most of these do, since most of these parse `$ZPOOL_IMPORT_PATH`), yield no entries, instead of garbage or whatever's on tap in the global buffer.

### How Has This Been Tested?
Ran the libzutil bits and zpool. zdb bits are trivial.

### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Performance enhancement (non-breaking change which improves efficiency) – zpool realloc-as-realloc instead of realloc-as-malloc-plus-memmove
- [x] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Library ABI change (libzfs, libzfs\_core, libnvpair, libuutil and libzfsbootenv)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:
- [x] My code follows the OpenZFS [code style requirements](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [x] I have updated the documentation accordingly.
- [x] I have read the [**contributing** document](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/openzfs/zfs/tree/master/tests) to cover my changes. – none should apply
- [ ] I have run the ZFS Test Suite with this change applied. – CI take my hand
- [x] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

**Review by jwk404 — CHANGES_REQUESTED** [review]
In order to prevent strtok from making future appearances, could I persuade you to add a change to make its presence a compile time error?

**Comment by nabijaczleweli** [comment]
How about doing
```sh
AM_CPPFLAGS += -D"strtok(...)=strtok(__VA_ARGS__) __attribute__((deprecated(\"Use strtok_r(3) instead!\")))"
```
which produces these errors:
```
  CC       zdb.o
zdb.c:3623:6: warning: 'strtok' is deprecated: Use strtok_r(3) instead! [-Wdeprecated-declarations]
        s = strtok(dup, ":");
            ^
<command line>:15:21: note: expanded from here
#define strtok(...) strtok(__VA_ARGS__) __attribute__((deprecated("Use strtok_r(3) instead!")))
                    ^
/usr/include/string.h:335:14: note: 'strtok' has been explicitly marked deprecated here
extern char *strtok (char *__restrict __s, const char *__restrict __delim)
             ^
<command line>:15:56: note: expanded from here
#define strtok(...) strtok(__VA_ARGS__) __attribute__((deprecated("Use strtok_r(3) instead!")))
                                                       ^
zdb.c:3623:6: error: expected ';' after expression
        s = strtok(dup, ":");
            ^
<command line>:15:41: note: expanded from here
#define strtok(...) strtok(__VA_ARGS__) __attribute__((deprecated("Use strtok_r(3) instead!")))
                                        ^
zdb.c:3623:6: error: 'deprecated' attribute cannot be applied to a statement
        s = strtok(dup, ":");
            ^               ~
<command line>:15:56: note: expanded from here
#define strtok(...) strtok(__VA_ARGS__) __attribute__((deprecated("Use strtok_r(3) instead!")))
                                                       ^
zdb.c:3632:24: error: too few arguments to function call, expected 3, have 2
        s = strtok_r(NULL, ":");
            ~~~~~~~~          ^
/usr/include/string.h:345:14: note: 'strtok_r' declared here
extern char *strtok_r (char *__restrict __s, const char *__restrict __delim,
             ^
zdb.c:3647:24: error: too few arguments to function call, expected 3, have 2
        s = strtok_r(NULL, ":");
            ~~~~~~~~          ^
/usr/include/string.h:345:14: note: 'strtok_r' declared here
extern char *strtok_r (char *__restrict __s, const char *__restrict __delim,
             ^
zdb.c:3651:31: error: too few arguments to function call, expected 3, have 2
        } else if (strtok_r(NULL, ":") != NULL) {
                   ~~~~~~~~          ^
/usr/include/string.h:345:14: note: 'strtok_r' declared here
extern char *strtok_r (char *__restrict __s, const char *__restrict __delim,
             ^
1 warning and 5 errors generated.
make: *** [Makefile:788: zdb.o] Error 1
```
?

File: udev/rules.d/.gitignore [hunk 1]
```diff
@@ -1,4 +1 @@
-69-vdev.rules
-60-zpool.rules
-60-zvol.rules
-90-zfs.rules
+*.rules
```
Comment (behlendorf) [inline udev/rules.d/.gitignore:1]: nit: a good but unrelated fix which seems to have ended up in these changes.  Otherwise, looks good to me.

**Review by behlendorf — APPROVED** [review]

File: udev/rules.d/.gitignore (hunk 1 above)
Comment (nabijaczleweli) [reply udev/rules.d/.gitignore:1]: removed

**Review by jwk404 — APPROVED** [review]
